### PR TITLE
schedule: check joint state when creating split/merge operator

### PR DESCRIPTION
**Pull request by HunDunDM**
### What problem does this PR solve?

* PD may try to merge/split the region in the joint state.

### What is changed and how it works?

* schedule: check joint state when creating split/merge operator

### Check List

Tests

- Unit test

### Release note

- No release note

### CODE REVIEW

File: server/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -78,7 +78,13 @@ func (c *RuleChecker) fixRange(region *core.RegionInfo) *operator.Operator {
 	if len(keys) == 0 {
 		return nil
 	}
-	return operator.CreateSplitRegionOperator("rule-split-region", region, 0, pdpb.CheckPolicy_USEKEY, keys)
```
Comment (HunDunDM) [inline server/schedule/checker/rule_checker.go:81]: @disksing Please confirm whether the rule checker needs to be updated.

Comment (disksing) [reply server/schedule/checker/rule_checker.go:81]: it's ok

File: server/schedule/checker/rule_checker.go [hunk 2]
```diff
@@ -78,7 +78,13 @@ func (c *RuleChecker) fixRange(region *core.RegionInfo) *operator.Operator {
...
+
+	op, err := operator.CreateSplitRegionOperator("rule-split-region", region, 0, pdpb.CheckPolicy_USEKEY, keys)
+	if err != nil {
+		return nil
```
Comment (disksing) [inline server/schedule/checker/rule_checker.go:84]: better to log the err

Comment (HunDunDM) [reply server/schedule/checker/rule_checker.go:85]: add a debug log

File: server/schedule/operator/create_operator.go [hunk 3]
```diff
@@ -95,7 +95,11 @@ func CreateMoveLeaderOperator(desc string, cluster opt.Cluster, region *core.Reg
 }
 
 // CreateSplitRegionOperator creates an operator that splits a region.
-func CreateSplitRegionOperator(desc string, region *core.RegionInfo, kind OpKind, policy pdpb.CheckPolicy, keys [][]byte) *Operator {
+func CreateSplitRegionOperator(desc string, region *core.RegionInfo, kind OpKind, policy pdpb.CheckPolicy, keys [][]byte) (*Operator, error) {
+	if core.IsInJointState(region.GetPeers()...) {
+		return nil, errors.Errorf("cannot split region which is in joint state")
```
Comment (rleungx) [inline server/schedule/operator/create_operator.go:100]: We need modify the error according to the new error standard.

**Review by disksing — APPROVED** [review]

File: server/schedule/operator/create_operator.go (hunk 3 above)
Comment (nolouch) [reply server/schedule/operator/create_operator.go:100]: cc @HunDunDM

Comment (HunDunDM) [reply server/schedule/operator/create_operator.go:100]: I think the `errors` under `operator` should be uniformly modified in another PR.

Comment (nolouch) [reply server/schedule/operator/create_operator.go:100]: Does the other errors do not meet the error code standard code?

Comment (HunDunDM) [reply server/schedule/operator/create_operator.go:100]: Yes, so this part of the modification requires cherry-pick.

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**